src/coreComponents/dataRepository/Utilities.cpp

```diff
@@ -0,0 +1,183 @@
+
+
+#include "common/MpiWrapper.hpp"
+#include "Utilities.hpp"
+#include "Group.hpp"
+
+
+#include <unordered_set>
+#include <unordered_map>
+
+namespace geosx
+{
+namespace dataRepository
+{
+
+void printMemoryAllocation( Group const & group, integer const indent, real64 const threshold )
+{
+  // static flag to keep track of whether or not a subgroup is the last one, and thus
+  // visually will terminate the tree.
+  static bool terminateBranch[64]{};
+
+  // keep track of the address of the groups that have been printed. Not the best
+  // way to do this, but doens't require infrastructure changes.
+  static std::unordered_set< void const * > groupsPrinted;
```

Comment:
Could it be
```suggestion
  static std::unordered_set< Group const * > groupsPrinted;
```
?

Reply:
I was trying to avoid allowing use of the pointer as a `Group *`... which would never happen...but the intent is to just be an address for purposes of comparison with no ability to call any facilities of `Group`. I have no problem changing it to `Group*` if you think it is better. I have no real preference.